pkg/eosclientgrpc/eosclientgrpc.go

```diff
@@ -0,0 +1,1347 @@
+	return ecl, nil
+}
+
+// New creates a new client with the given options.
+func New(opt *Options) *Client {
+	tlog := logger.New().With().Int("pid", os.Getpid()).Logger()
+
+	fmt.Printf("--- opt '%+v'\n", opt)
+	opt.init()
+	c := new(Client)
+	c.opt = opt
+
+	tctx := appctx.WithLogger(context.Background(), &tlog)
+
+	// Let's be successful if the ping was ok. This is an initialization phase
+	// and we enforce the server to be up
+	// This will likely improve as soon as the behaviour of grpc is understood
+	// in the case of server restarts or failures
+	ccl, err := newgrpc(tctx, opt)
+	if err != nil {
+		return nil
+	}
+	c.cl = ccl
+
+	// Some connection tests, useful for logging in this dev phase
+
+	fmt.Printf("--- Going to stat '%s'\n", "/eos")
+	frep, err := c.GetFileInfoByPath(tctx, "furano", "/eos")
+	if err != nil {
+		fmt.Printf("--- GetFileInfoByPath '%s' failed with err '%s'\n", "/eos", err)
+		//	return nil
+	} else {
+		fmt.Printf("--- GetFileInfoByPath to '%s' gave response '%s'\n", "/eos", frep.File)
+	}
+
+	fmt.Printf("--- Going to stat '%s'\n", "/eos-idonotexist")
+	frep1, err := c.GetFileInfoByPath(tctx, "furano", "/eos-idonotexist")
+	if err != nil {
+		fmt.Printf("--- GetFileInfoByPath '%s' failed with err '%s'\n", "/eos-idonotexist", err)
+		//	return nil
+	} else {
+		fmt.Printf("--- GetFileInfoByPath to '%s' gave response '%s'\n", "/eos-idonotexist", frep1.File)
+	}
+
+	fmt.Printf("--- Going to list '%s'\n", "/eos")
+	lrep, err := c.List(context.Background(), "furano", "/eos")
+	if err != nil {
+		fmt.Printf("--- List '%s' failed with err '%s'\n", "/eos", err)
+		//	return nil
+	} else {
+		fmt.Printf("--- List to '%s' gave %d entries\n", "/eos", len(lrep))
+	}
+
+	fmt.Printf("--- Going to getACLForPath '%s'\n", "/eos/cms")
+	arep, err := c.getACLForPath(context.Background(), "furano", "/eos/cms")
+	if err != nil {
+		fmt.Printf("--- getACLForPath '%s' failed with err '%s'\n", "/eos/cms", err)
+		//	return nil
+	} else {
+		for i, s := range arep.Entries {
+			fmt.Printf("--- getACLForPath to '%s' gave %d:'%s'\n", "/eos/cms", i, s)
+		}
+	}
+
+	return c
+}
+
+func (c *Client) getUnixUser(username string) (*gouser.User, error) {
+	if c.opt.ForceSingleUserMode {
+		username = c.opt.SingleUsername
+	}
+	return gouser.Lookup(username)
+}
+
+// AddACL adds an new acl to EOS with the given aclType.
+func (c *Client) AddACL(ctx context.Context, username, path string, a *acl.Entry) error {
+	acls, err := c.getACLForPath(ctx, username, path)
+	if err != nil {
+		return err
+	}
+
+	// since EOS Citrine ACLs are is stored with uid, we need to convert username to uid
+	// only for users.
+	if a.Type == acl.TypeUser {
+		a.Qualifier, err = getUID(a.Qualifier)
+		if err != nil {
+			return err
+		}
+	}
+	err = acls.SetEntry(a.Type, a.Qualifier, a.Permissions)
+	if err != nil {
+		return err
+	}
+	sysACL := acls.Serialize()
+
+	// Stuff filename, uid, gid into the MDRequest type
+	rq := new(erpc.NSRequest)
```

Comment:
This does the same thing as `initNSRequest`, so we can just call that. Ditto for other methods.